src/formatter/mod.rs

```diff
@@ -217,16 +218,21 @@ impl<'a> DisplayList<'a> {
                     } else {
                         false
                     };
+                    let mut ended = false;
```

Comment:
can you document this variable? It's a bit unique.

I also feel like there must be a way to make it simpler. The `range.0` is always the result of `acc.0`, so basically the first `take()`, and the `range.1` should be chainable `Option` on last element of `take_while` so if you put the iterator as a mutable variable, you should be able to do `iter.next().or(last)`.

The refactor is optional, and I'm happy to land this as-is with a comment explaining the variable and its role.

Reply:
This `ended` specifies that it will end in the next character so it will return until the next one to the final condition. In order to measure this last character, when end condition is true, in an efficient way. 

I will try to document it.

Reply:
I think I understand the role, but I had to run an iterator in my head to derive to that conclusion so a doc string would help.